File: Java/Aula18/src/aula18/ContaBanco.java
package aula18;
public class ContaBanco {
    private int numConta;
    private String tipo;
    private String dono;
    private float saldo;
    private boolean status;

    public int getNumConta() {
        return numConta;
    }

    public void setNumConta(int numConta) {
        this.numConta = numConta;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getDono() {
        return dono;
    }

    public void setDono(String dono) {
        this.dono = dono;
    }

    public float getSaldo() {
        return saldo;
    }

    public void setSaldo(float saldo) {
        this.saldo = saldo;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    
    
    public ContaBanco(int numConta, String tipo, String dono, float saldo, boolean status) {
        this.numConta = numConta;
        this.tipo = tipo;
        this.dono = dono;
        this.saldo = 0;
        this.status = false;
    }
    public ContaBanco(){
        this.saldo = 0;
        this.status = false;
    }
    
    public void abrirConta(String tipo){
        if (tipo.equals("CC")){
            this.setSaldo(50);
        } else if (tipo.equals("CP")){
            this.setSaldo(150);
        }
        this.setTipo(tipo);
        this.setStatus(true);
    }
    
    public void fecharConta(){
        if (this.getSaldo() != 0){
            System.out.println("Retire TODO o Saldo para Finalizar a Conta");
            return;
        }
        this.setStatus(false);
    }
 
    public void depositar(float deposito){
        if (this.status == false){
            System.out.println("A Conta foi Encerrada. Não é possível depositar.");
            return;
        }
        float var = this.getSaldo() + deposito;
        this.setSaldo(var);
    }
    
    public void sacar(float saque){
        if (this.status == false){
            System.out.println("A Conta foi Encerrada. Não há dinheiro nem é possível sacar.");
            return;
        }
        if (this.getSaldo() < saque){
            System.out.println("Saque maior que Saldo. Saque Indisponível.");
            return;
        }
        float var = this.getSaldo() - saque;
        this.setSaldo(var);
    }
    
    public void pagarMensalidade(){
        if (this.isStatus() == false){
            System.out.println("Conta Encerrada. Nenhuma mensalidade será cobrada.");
            return;
        }
        if (this.getTipo().equals("CC")){
            if (getSaldo() < 12){
                System.out.println("Não é possível debitar a mensalidade da Conta Corrente.");
                return;
            }
            float var = getSaldo() - 12f;
            setSaldo(var);
            System.out.println("Mensalidade da Conta Corrente debitada.");
            return;
        }
        if (this.getTipo().equals("CP")){
            if (getSaldo() < 20){
                System.out.println("Não é possível debitar a mensalidade da Conta Poupança");
                return;
            }
            float var = getSaldo() - 20f;
            setSaldo(var);
            System.out.println("Mensalidade da Conta Poupança debitada.");
            return;
        }
    }
    
    public void estadoAtual(){
        System.out.println("==============================================");
        System.out.println("Número da Conta: " + this.getNumConta());
        System.out.println("Status: " + this.isStatus());
        System.out.println("Tipo da Conta: " + this.getTipo());
        System.out.println("Titular: " + this.getDono());
        System.out.println("Saldo: " + this.getSaldo());
        System.out.println("----------------------------------------------");
    }
}

File: PHP/Aula31/ContaBanco.php
<?php

class ContaBanco {
    
    public $numConta;
    protected $tipo;
    private $dono;
    private $saldo;
    private $status;

    public function __construct() {
        $this->setSaldo(0);
        $this->setStatus(false);
        echo "<p>Conta criada com sucesso.</p>";
    }

    public function getNumConta() {
        return $this->numConta;
    }
    public function setNumConta($numConta) {
        $this->numConta = $numConta;
    }

    public function getTipo() {
        return $this->tipo;
    }
    public function setTipo($tipo) {
        $this->tipo = $tipo;
    }

    public function getDono() {
        return $this->dono;
    }
    public function setDono($dono) {
        $this->dono = $dono;
    }

    public function getSaldo() {
        return $this->saldo;
    }
    public function setSaldo($saldo) {
        $this->saldo = $saldo;
    }

    public function getStatus() {
        return $this->status;
    }
    public function setStatus($status) {
        $this->status = $status;
    }

    public function abrirConta($tipo) {
        $this->setTipo($tipo);
        $this->setStatus(true);
        
        if ($tipo == "CC") {
            $this->setSaldo(50);
        } elseif ($tipo == "CP") {
            $this->setSaldo(150);
        }
    }

    public function fecharConta() {
        if ($this->getSaldo() > 0) {
            echo "<p>Esvazie a Conta primeiro</p>";
        } elseif ($this->getSaldo() < 0) {
            echo "<p>Conta está em débito. Impossível encerrar-la.</p>";
        } else {
            $this->setStatus(false);
            echo "<p>Conta de {$this->getDono()} fechada com sucesso.</p>";
        }
    }

    public function depositar($v) {
        if ($this->getStatus()) {
            $this->setSaldo($this->getSaldo() + $v);
            echo "<p>Depósito de R$$v na conta de {$this->getDono()}.</p>";
        } else {
            echo "<p>Conta encerrada. Não consigo depositar.</p>";
        }
    }

    public function sacar($v) {
        if ($this->getStatus()) {
            if ($this->getSaldo() >= $v) {
                $this->setSaldo($this->getSaldo() - $v);
                echo "<p>Saque de R$$v autorizado na conta de {$this->getDono()}.</p>";
            } else {
                echo "<p>Saldo insuficiente para saque.</p>";
            }
        } else {
            echo "<p>Não posso sacar de uma conta encerrada.</p>";
        }
    }

    public function pagarMensal() {
        if ($this->getTipo() == "CC") {
            $v = 12;
        } elseif ($this->getTipo() == "CP") {
            $v = 20;
        }
        
        if ($this->getStatus()) {
            if ($this->getSaldo() > $v) {
                $this->setSaldo($this->getSaldo() - $v);
                echo "<p>Mensalidade de R$ $v debitada na conta de {$this->getDono()}.</p>";
            } else {
                echo "<p>Saldo insuficiente.</p>";
            }
        } else {
            echo "<p>Problemas com a conta. Não posso cobrar.</p>";
        }
    }
}
?>